src/adapter/graphql/tests/tests/test_gql_account_flows_configs.rs

```diff
@@ -0,0 +1,765 @@
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+use std::sync::Arc;
+
+use async_graphql::value;
+use chrono::Duration;
+use dill::Component;
+use event_bus::EventBus;
+use indoc::indoc;
+use kamu::testing::{
+    MetadataFactory,
+    MockAuthenticationService,
+    MockDatasetActionAuthorizer,
+    MockDatasetChangesService,
+    MockDependencyGraphRepository,
+    MockPollingIngestService,
+    MockTransformService,
+};
+use kamu::{set_random_jwt_secret, DatasetRepositoryLocalFs, DependencyGraphServiceInMemory};
+use kamu_core::auth::AccountInfo;
+use kamu_core::*;
+use kamu_flow_system::FlowServiceRunConfig;
+use kamu_flow_system_inmem::{FlowConfigurationEventStoreInMem, FlowEventStoreInMem};
+use kamu_flow_system_services::{FlowConfigurationServiceImpl, FlowServiceImpl};
+use kamu_task_system_inmem::TaskSystemEventStoreInMemory;
+use kamu_task_system_services::TaskSchedulerImpl;
+use mockall::predicate::eq;
+use opendatafabric::{AccountName, DatasetAlias, DatasetID, DatasetKind, DatasetName};
+
+use crate::utils::{add_custom_logged_in_account, authentication_catalogs};
+
+#[test_log::test(tokio::test)]
+async fn test_list_account_flows() {
+    let foo_account_name = AccountName::new_unchecked("foo");
+    let foo_dataset_name = DatasetName::new_unchecked("foo");
+    let foo_dataset_alias =
+        DatasetAlias::new(Some(foo_account_name.clone()), foo_dataset_name.clone());
+    let mut mock_authentication_service = MockAuthenticationService::new();
+    mock_authentication_service
+        .expect_find_account_info_by_name()
+        .with(eq(foo_account_name.clone()))
+        .returning(|_| {
+            Ok(Some(AccountInfo {
+                account_id: "foo_id".to_string(),
+                account_name: AccountName::new_unchecked("foo"),
+                account_type: auth::AccountType::User,
+                display_name: "foo".to_string(),
+                avatar_url: None,
+                is_admin: false,
+            }))
+        });
+    let mock_dataset_action_authorizer = MockDatasetActionAuthorizer::allowing();
+    let harness = FlowConfigHarness::with_overrides(FlowRunsHarnessOverrides {
+        transform_service_mock: Some(MockTransformService::with_set_transform()),
+        polling_service_mock: Some(MockPollingIngestService::with_active_polling_source()),
+        mock_authentication_service: Some(mock_authentication_service),
+        mock_dataset_action_authorizer: Some(mock_dataset_action_authorizer),
+        ..Default::default()
+    });
+
+    let create_result = harness.create_root_dataset(foo_dataset_alias).await;
+    let schema = kamu_adapter_graphql::schema_quiet();
+    let foo_catalog_authorized =
+        add_custom_logged_in_account(&harness.catalog_base, foo_account_name.clone(), false);
```

Comment:
Can't you just pass the name of the account into harness constructor, and configure catalog with existing mechanisms? It looks like `add_custom_logged_in_account` is not much of a value, and can be expressed with existing facitilities.

Reply:
Removed completely by mock service